src/Toolkit/PubSub/V1/SubscriberApi.php

```diff
@@ -0,0 +1,901 @@
+            ['call_credentials_callback' => $this->createCredentialsCallback()]);
+    }
+
+    /**
+     * Deletes an existing subscription. All pending messages in the subscription
+     * are immediately dropped. Calls to `Pull` after deletion will return
+     * `NOT_FOUND`. After a subscription is deleted, a new one may be created with
+     * the same name, but the new one has no association with the old
+     * subscription, or its topic unless the same topic is specified.
+     *
+     * Sample code:
+     * <pre><code>
+     * try {
+     *     $subscriberApi = new SubscriberApi();
+     *     $formattedSubscription = SubscriberApi::formatSubscriptionName("[PROJECT]", "[SUBSCRIPTION]");
+     *     $subscriberApi->deleteSubscription($formattedSubscription);
+     * } finally {
+     *     if (isset($subscriberApi)) {
+     *         $subscriberApi->close();
+     *     }
+     * }
+     * </code></pre>
+     *
+     * @param string $subscription The subscription to delete.
+     * @param array  $optionalArgs {
+     *                             Optional. There are no optional parameters for this method yet;
+     *                             this $optionalArgs parameter reserves a spot for future ones.
+     *                             }
+     * @param array  $callSettings {
+     *                             Optional.
+     *
+     *    @var Google\GAX\RetrySettings $retrySettings
+     *          Retry settings to use for this call. If present, then
+     *          $timeout is ignored.
+     *    @var int $timeoutMillis
+     *          Timeout to use for this call. Only used if $retrySettings
+     *          is not set.
+     * }
+     *
+     * @throws Google\GAX\ApiException if the remote call fails
+     */
+    public function deleteSubscription($subscription, $optionalArgs = [], $callSettings = [])
+    {
+        $request = new DeleteSubscriptionRequest();
+        $request->setSubscription($subscription);
+
+        $mergedSettings = $this->defaultCallSettings['deleteSubscription']->merge(
+            new CallSettings($callSettings)
+        );
+        $callable = ApiCallable::createApiCall(
+            $this->stub,
+            'DeleteSubscription',
+            $mergedSettings,
+            $this->descriptors['deleteSubscription']
+        );
+
+        return $callable(
+            $request,
+            [],
+            ['call_credentials_callback' => $this->createCredentialsCallback()]);
+    }
+
+    /**
+     * Modifies the ack deadline for a specific message. This method is useful
+     * to indicate that more time is needed to process a message by the
+     * subscriber, or to make the message available for redelivery if the
+     * processing was interrupted.
+     *
+     * Sample code:
+     * <pre><code>
+     * try {
+     *     $subscriberApi = new SubscriberApi();
+     *     $formattedSubscription = SubscriberApi::formatSubscriptionName("[PROJECT]", "[SUBSCRIPTION]");
+     *     $ackIds = [];
```

Comment:
So I did some research on push messaging, and it turns out that messages received through push do not contain an ackId, and thus there is no ackId to pass to modifyAckDeadline in that scenario. https://cloud.google.com/pubsub/subscriber#receive_push Thus, it seems that modifyAckDeadline only works with the pull scenario.